### ISSUE TITLE
Broken (more like undefined) PubID for committee document with stage

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-iso#1195]
Is this a PubID issue?

<img width="647" alt="Screenshot 2024-07-25 at 10 04 39 PM" src="https://github.com/user-attachments/assets/c502a369-fe86-4829-99e7-d782702ee953">

<img width="623" alt="Screenshot 2024-07-25 at 10 04 29 PM" src="https://github.com/user-attachments/assets/683c5d12-5d5e-4c09-9685-049c2ba76705">

Because the docstage being 20/20 the PubID is now wrong.

Right now it is shown as:
* `ISO/TCWD 184/SC 4/WG 12 N11498`

Technically we have no defined the PubID for N-documents, because usually N-documents are replaced per version.

Taking the pattern of "{publisher}/{stage} {number}", perhaps it should be:
* `ISO/TC 184/SC 4/WG 12/WD N11498`

Metadata:
```
= Validation report for conversion of ISO 10303 parts from PDF form to a model-based form
:docnumber: 11498
:edition: 1
:revdate: 2024-07-17
:copyright-year: 2024
:language: en
:title-intro-en: SMRL Innovation
:title-intro-fr: Innovation de SMRL
:title-main-en: Validation report for conversion of ISO 10303 parts from PDF form to a model-based form
:title-main-fr: Rapport de validation pour la conversion des pièces ISO 10303 d'un formulaire PDF à un formulaire basé sur un modèle
:doctype: committee-document
:docstage: 20
:docsubstage: 20
:secretariat: ANSI
:technical-committee-number: 184
:technical-committee: Industrial Automation Systems and Integration
:subcommittee-number: 4
:subcommittee: Industrial Data
:workgroup-number: 12
:workgroup-type: WG
:workgroup: STEP product modelling and resources
```

**Comment by opoudjis** [comment]
The identifier generation for ISO identifiers is completely delegated to pubid, by design. I have no idea how N is even ending up in the identifier. This is for @mico

**Comment by opoudjis** [comment]
This is being generated with the parameters:

````
{:number=>"11498", :type=>:tc, :publisher=>"ISO", :stage=>"20.20", :sctype=>"SC", :scnumber=>"4", :tctype=>"TC", :tcnumber=>"184", :wgtype=>"WG", :wgnumber=>"12"}
````

**Comment by mico** [comment]
> This is being generated with the parameters:
> 
> ```
> {:number=>"11498", :type=>:tc, :publisher=>"ISO", :stage=>"20.20", :sctype=>"SC", :scnumber=>"4", :tctype=>"TC", :tcnumber=>"184", :wgtype=>"WG", :wgnumber=>"12"}
> ```

@opoudjis yes, with this input we get stage in output.
@ronaldtse should we ignore the stage and produce output without, or we should add stage to output? (it will be first known TC identifier with stage)

**Comment by ronaldtse** [comment]
@opoudjis @mico there are 2 types of ISO IDs now:

* ISO standard IDs
* ISO N-document IDs

These two types of IDs need to be handled separately.

ISO N-document IDs do not show stages.

**Comment by mico** [comment]
> * ISO N-document IDs

@ronaldtse is there other N documents except TC documents? (ISO/TC)

**Comment by ronaldtse** [comment]
> @ronaldtse is there other N documents except TC documents? (ISO/TC)

No. TC documents the "N documents" because they have an "N-number".
